File: JAVA/BinaryTrees/LeftView.java
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res = new LinkedList<>();
        if(root == null) return res;
        LinkedList<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while(!queue.isEmpty()){
            int size = queue.size();
            for(int i = 0; i < size; i++){
                TreeNode node = queue.poll();
                if(node.left != null) queue.add(node.left);
                if(node.right != null) queue.add(node.right);
                if(i == 0) res.add(node.val);
                
            }
        }
        
        return res;
    }
}

File: C++/Binary Trees/Revision_2025/Left_View.cpp
/*

class Node {
public:
    int data;
    Node* left;
    Node* right;

    Node(int val) {
        data = val;
        left = nullptr;
        right = nullptr;
    }
};
 */

class Solution {
  public:
    vector<int> leftView(Node *root) {
        // code here
        if(!root)
            return {};
            
        vector<int> res;
        
        queue<Node*> q;
        q.push({root});
        
        while(!q.empty()){
            int len = q.size();
            
            for(int i = 0; i < len; i++){
                Node* currNode = q.front();
                q.pop();
                
                if(i==0)
                    res.push_back(currNode->data);
                
                if(currNode->left)
                    q.push(currNode->left);
                
                if(currNode->right)
                    q.push(currNode->right);
            }
        }
        
        return res;
    }
};
